HLTrigger/Muon/src/HLTScoutingMuonProducer.cc

```diff
@@ -83,36 +86,121 @@ void HLTScoutingMuonProducer::produce(edm::StreamID sid, edm::Event & iEvent,
         return;
     }
 
+    //get vertices
+    Handle<reco::VertexCollection> vertexCollection;
+    if(iEvent.getByToken(vertexCollection_, vertexCollection)){
+      for(auto &vtx : *vertexCollection){
+	if ( !vtx.isValid() ) continue ;
+	outVertices->emplace_back(
+				  vtx.x(), vtx.y(), vtx.z(), vtx.zError(), vtx.xError(), vtx.yError(), vtx.tracksSize(), vtx.chi2(), vtx.ndof()
+				  );
+      }
+    }
+
+    std::pair<reco::RecoChargedCandidate,reco::RecoChargedCandidate> ivtxMuPair;
+    std::vector<std::pair<reco::RecoChargedCandidate,reco::RecoChargedCandidate> > vtxMuPair;
+    vtxMuPair.clear();
+    
+    //get displaced vertices
+    Handle<reco::VertexCollection> displacedvertexCollection;
+    if(iEvent.getByToken(displacedvertexCollection_, displacedvertexCollection)){
+      
+      for(auto &dispvtx : *displacedvertexCollection){
+	if ( !dispvtx.isValid() ) continue ;
+	float vtxProb = 0.0;
+	if( (dispvtx.chi2()>=0.0) && (dispvtx.ndof()>0) ) vtxProb = TMath::Prob(dispvtx.chi2(), dispvtx.ndof() );
+	if (vtxProb < minVtxProbCut) continue;
+	
+	// Get the 2 tracks associated to displaced vertex
+	reco::Vertex::trackRef_iterator trackIt =  dispvtx.tracks_begin();
+	reco::TrackRef vertextkRef1 =  (*trackIt).castTo<reco::TrackRef>() ;
+	trackIt++;
+	reco::TrackRef vertextkRef2 =  (*trackIt).castTo<reco::TrackRef>();
+	
+	// Get the muons associated with the tracks
+	int iFoundRefs = 0;
+	for (reco::RecoChargedCandidateCollection::const_iterator cand=ChargedCandidateCollection->begin(); cand!=ChargedCandidateCollection->end(); cand++) {
+	  reco::TrackRef tkRef = cand->get<reco::TrackRef>();
+	  if(tkRef == vertextkRef1) {ivtxMuPair.first= (*cand); iFoundRefs++ ;}
+	  if(tkRef == vertextkRef2) {ivtxMuPair.second= (*cand); iFoundRefs++ ;}
+	}
+	if (iFoundRefs<2) continue;
+	vtxMuPair.push_back(ivtxMuPair);
+	
+	dispVertices->emplace_back(
+				   dispvtx.x(), dispvtx.y(), dispvtx.z(), 
+				   dispvtx.zError(), dispvtx.xError(), 
+				   dispvtx.yError(), dispvtx.tracksSize(), 
+				   dispvtx.chi2(), dispvtx.ndof()
+				   );
+	
+      }
+    }
+    
     // Produce muons
+    std::vector<int> vtxInd;
+    vtxInd.clear();
```

Comment:
There is no need to call `clear` hear since `vtxInd` is guaranteed to be empty at this point.